### PR TITLE
Adds support for a new DBStateChanged event 

### LINKED ISSUE #1176: Add WebHook notifications for bucket online/offline
Bucket online offline events should be added to the WebHooks service.

### PR DESCRIPTION
**Pull request by ajres**
Adds support for a new DBStateChanged event fired when a DB changes state to online or offline

fixes #1176

### CODE REVIEW

File: src/github.com/couchbase/sync_gateway/db/event_handler.go [hunk 1]
```diff
@@ -93,6 +93,15 @@ func (wh *Webhook) HandleEvent(event Event) {
 		}
 		contentType = "application/json"
 		payload = bytes.NewBuffer(jsonOut)
+	case *DBStateChangeEvent:
+		// for DBStateChangeEvent, post document body
```
Comment (adamcfraser) [inline src/github.com/couchbase/sync_gateway/db/event_handler.go]: Maybe clarify in this comment what the document body actually represents?

File: src/github.com/couchbase/sync_gateway/db/event.go [hunk 2]
```diff
@@ -25,18 +26,18 @@ type Event interface {
...
+//type EventImpl struct {
+//	eventType EventType
+//}
 
-func (e *EventImpl) EventType() EventType {
-	return e.eventType
-}
+//func (e *EventImpl) EventType() EventType {
+//	return e.eventType
+//}
```
Comment (adamcfraser) [inline src/github.com/couchbase/sync_gateway/db/event.go]: Might as well take this right out.

**Comment by adamcfraser** [comment]
I don't think it makes sense to merge this before #1013.

**Comment by ajres** [comment]
Rebased with latest master and added calls from online/offline to RaiseDBStateChangeEvent

**Comment by adamcfraser** [comment]
The changes look good, but the unit test doesn't seem to provide a lot of coverage or verification that things are working as expected.  At minimum it could be verifying the payload sent by the webhook, and possibly some negative tests.  It would be better if we had a unit test to actually verify webhook notification on db state change, but that can be left as a functional test for now.

**Comment by ajres** [comment]
Added payload verification to the TestingHandler
I'm not sure what negative tests would add value, the DBStateChangeEvent values are all passed in by the test as strings, it feels like we would just be testing the test?

**Comment by adamcfraser** [comment]
Build failures appear to be the race conditions fixed elsewhere.  Looks ready to merge.

**Pull request merged**